Fix RPC_Patch_V1_02 (#513)
Azure/azure-openapi-validator: docs/un-supported-patch-properties.md

## Output Message

> The patch operation body parameter schema should not contain top level "id", "name", "type", "location" as writable properties. This is because these properties are expected to be either readOnly or immutable.

Reviewer: ```suggestion
The patch operation body parameter schema should not contain top level "id", "name", "type", or "location" as writable properties. This is because these properties are expected to be either readOnly or immutable.
```

Author: hmm including them as top-level would not require them to be writable.

Author: The properties should either not exist in the patch request body or if they are specified, they must be marked as readOnly or immutable

> The patch operation body parameter schema should not contain top level "id", "name", "type", "location" as writable properties. This is because these properties are expected to be either readOnly or immutable.

Reviewer: ```suggestion
The patch operation body parameter schema should not contain top-level "id", "name", "type", "location" as writable properties. To fix this, either remove the offending properties, or mark them as readOnly or immutable.
```

I think this would convey what is going a bit more naturally, based on our discussion.

Reviewer 2: Could also put it in quotes

Author: I felt the "How to ix the violation was where we tell what to do to fix the violation so left the Output message to only reflect an explanation of the problem

Author: But I will change it anyway.

Author: Which part in quotes Brett?

Reviewer 2: I meant to put that reply on https://github.com/Azure/azure-openapi-validator/pull/513#discussion_r1196076146

## How to fix the violation

> Consider either removing the top level properties - id, name and type, from the patch request body or mark them as readOnly. For the top level location property for tracked resources, consider either removing it from the request body of the Patch operation or mark it as immutable using the x-ms-mutability property and values as "create" and "read".

Reviewer: ```suggestion
Consider either removing the top level properties - id, name and type, from the patch request body, or mark them as readOnly. For the top level location property for tracked resources, consider either removing it from the request body of the Patch operation or mark it as immutable using the x-ms-mutability property and values as "create" and "read".
```

Author: fixed

> Consider either removing the top level properties - id, name and type, from the patch request body or mark them as readOnly. For the top level location property for tracked resources, consider either removing it from the request body of the Patch operation or mark it as immutable using the x-ms-mutability property and values as "create" and "read".

Reviewer: What does `for tracked resources` mean here? Can there be a top-level "location" property for untracked resources?

Author: "location" property can only exist for tracked resources. Tracked resources are resources that are stored by ARM in its registry. A resource is a representation of an object that follows the REST semantics

Author: I edited the description to make it convey the meaning better
